lib/puppet/type/tcp_conn_validator.rb

```diff
@@ -0,0 +1,44 @@
+Puppet::Type.newtype(:tcp_conn_validator) do
+
+  @doc = "Verify that a TCP connection can be successfully established between a node
+          and the expected  server.  Its primary use is as a precondition to
+          prevent configuration changes from being applied if the server cannot be
+          reached, but it could potentially be used for other purposes such as monitoring."
+
+  ensurable do
+    defaultvalues
+    defaultto :present
+  end
+
+  newparam(:name, :namevar => true) do
+    desc 'An arbitrary name used as the identity of the resource.'
+  end
+
+  newparam(:server) do
+    desc 'An array containing DNS names or IP addresses of the server where the expected service should be running.'
+    defaultto '127.0.0.1'
+    munge do |value|
+      Array(value).first
+    end
+  end
+
+  newparam(:port) do
+    desc 'The port that the server should be listening on.'
+    defaultto '22'
```

Comment:
Is it possible to have a meaningful default for the port, or should the user have to specify a port to connect to every time?